lib/utils/ver.go

```diff
@@ -17,35 +17,34 @@ limitations under the License.
 package utils
 
 import (
-	"github.com/coreos/go-semver/semver"
+	"fmt"
 
 	"github.com/gravitational/trace"
+
+	"github.com/coreos/go-semver/semver"
 )
 
-// CheckVersions compares client and server versions
-// and makes sure they are compatible using Teleport convention
-func CheckVersions(clientVersion, serverVersion string) error {
+// CheckVersions compares client and server versions and makes sure that the
+// client version is greater than or equal to the minimum version supported
+// by the server.
+func CheckVersions(clientVersion string, minClientVersion string) error {
 	clientSemver, err := semver.NewVersion(clientVersion)
 	if err != nil {
 		return trace.Wrap(err,
 			"unsupported version format, need semver format: %q, e.g 1.0.0", clientVersion)
 	}
 
-	serverSemver, err := semver.NewVersion(serverVersion)
+	minClientSemver, err := semver.NewVersion(minClientVersion)
```

Comment:
in backported version, should not you handle the case when server does not communicate the version yet and fallback to the old logic? As is it this commit could break 3.0.1 to 3.0.0 communications?

Reply:
This comment applies to master as well?

Reply:
Won't break 3.0.2 to 3.0.0 communication because the check as simply been removed if the server does not advertise a minimum version.